File: src/main/java/ru/nsu/fit/team_project/model/commands/AddFieldCommand.java
package ru.nsu.fit.team_project.model.commands;

import ru.nsu.fit.team_project.model.MObject;
import ru.nsu.fit.team_project.model.Model;
import ru.nsu.fit.team_project.model.fields.Field;

import java.util.UUID;

public class AddFieldCommand extends Command {
    UUID objectID;
    UUID fieldID = UUID.randomUUID();
    String fieldName;
    String fieldType;

    public AddFieldCommand(UUID objectId, String fieldName, String fieldType) {
        super("AddFieldCommand");
        this.objectID = objectId;
        this.fieldName = fieldName;
        this.fieldType = fieldType;
    }

    @Override
    public void execute(Model model) {
        MObject object = model.getObjects()
                              .stream()
                              .filter(o -> o.getId().equals(objectID))
                              .findAny()
                              .orElseThrow();

        object.addField(new Field(fieldID, fieldName));
    }
}

File: src/jvmMain/kotlin/data/model/commands/AddFieldCommand.kt
package data.model.commands

import data.model.Field
import data.model.Model
import java.util.*

class AddFieldCommand(
    private val objectId: UUID,
    private val fieldId: UUID,
    private val fieldName: String,
    private val fieldType: String
) : Command(COMMAND_NAME) {

    private companion object {

        private const val COMMAND_NAME = "AddField"
    }

    override fun execute(model: Model) {
        model.objects
            .stream()
            .filter { o -> o.id == objectId }
            .findAny()
            .orElseThrow()
            .addField(
                Field(
                    fieldId,
                    fieldName
                )
            )
    }
}